### PR TITLE
feat(spanner): support pg jsonb

### PR DESCRIPTION
**Pull request by NivedhaSenthil**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [ ] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [ ] Update code documentation if necessary.

closes: #<issue_number_goes_here>

### CODE REVIEW

File: google-cloud-spanner/acceptance/spanner/client/params/pgjsonb_test.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  let(:db) { spanner_pg_client }
+  let(:json_params) { { "venue" => "abc", "rating" => 10 } }
+  let :json_array_params do
+    3.times.map do |i|
+      { "venue" => "abc-#{i}", "rating" => 10 + i }
+    end
+  end
+
+  it "queries and returns a string parameter" do
+    skip if emulator_enabled?
```
Comment (dazuma) [inline google-cloud-spanner/acceptance/spanner/client/params/pgjsonb_test.rb:27]: Totally optional, feel free to ignore (or argue that it's better for it to be explicit in every test):

This seems to be consistent across all tests. You could avoid repeating it in every test with:

```ruby
before do
  skip if emulator_enabled?
end
```

File: google-cloud-spanner/acceptance/spanner/client/types/pgjsonb_test.rb [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  let :json_array_params do
+    3.times.map do |i|
+      { "venue" => "abc-#{i}", "rating" => 10 + i }
+    end
+  end
+
+  it "writes and reads json" do
+    skip if emulator_enabled?
+
+    id = SecureRandom.int64
```
Comment (dazuma) [inline google-cloud-spanner/acceptance/spanner/client/types/pgjsonb_test.rb:30]: Totally optional, feel free to ignore:

This seems to be consistent across all tests. You could avoid repeating it with:

```ruby
  let(:id) { SecureRandom.int64 }
```

**Review by dazuma — APPROVED** [review]
LGTM

File: google-cloud-spanner/acceptance/spanner/client/params/pgjsonb_test.rb (hunk 1 above)
Comment (NivedhaSenthil) [reply google-cloud-spanner/acceptance/spanner/client/params/pgjsonb_test.rb:27]: Thanks Daniel, have updated accordingly 👍

File: google-cloud-spanner/acceptance/spanner/client/types/pgjsonb_test.rb (hunk 2 above)
Comment (NivedhaSenthil) [reply google-cloud-spanner/acceptance/spanner/client/types/pgjsonb_test.rb:32]: I would prefer leaving this as this creates random id for each tests to makes sure that they are isolated and not the same row is used for every test.

Comment (NivedhaSenthil) [reply google-cloud-spanner/acceptance/spanner/client/types/pgjsonb_test.rb:32]: Just realised the let block is executed each time we access it not just once 👍 Updated the tests.

**Pull request merged**